Assets/Scripts/Collectables/HealthBasedDropper.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Actor;
+using UnityEngine;
+
+namespace Collectables
+{
+    /// <summary>
+    /// A dropper that drops drops based on the health of the entity it's bound to.
+    /// </summary>
+    public class HealthBasedDropper : Dropper
+    {
+        /// <summary>
+        /// The entity that this dropper is attached to. Used to keep track of health.
+        /// </summary>
+        public Living entity;
+        
+        /// <summary>
+        /// The health points at which to drop drops.
+        /// </summary>
+        public int[] healthMarkers;
```

Comment:
I think this logic, the logic that drops items based-off of health, can be done more generally. Given that we have many dropper classes, we can create a `IDispenser` that dispense any `Dropper` class. Otherwise we will have to create Dropper classes for each type of Dropper and that seems to be more difficult to manage.

IDEA:
Dropper interface now keeps track of the health threshold for when to drop an item.
Generate a IDispenser that scans it's child GameObject with Dropper objects sorted and bucket them by order of health thresholds. Then dispense / spawn those items at the appropriate times.

Reply:
Refactored to have a single dropper that can take in a list of pickups to drop.